diesel/src/query_builder/mod.rs

```diff
@@ -55,7 +56,11 @@ pub use self::sql_query::{BoxedSqlQuery, SqlQuery};
 pub use self::upsert::on_conflict_target_decorations::DecoratableTarget;
 
 #[doc(inline)]
-pub use self::update_statement::changeset::AsChangeset;
+pub use self::uncorrelated_column::UncorrelatedColumn;
+#[doc(inline)]
+pub use self::update_statement::changeset::{
+    AsChangeset, AssignableExpression, AssignmentTarget, ColumnWrapperForUpdate,
+};
```

Comment:
None of those types should be part of our public API, as they are only internal helper types. 
For those types that are required for implementing the `table!` macro: Reexport them via `diesel::internal::table_macro` as `#[doc(hidden)]` item.